pkg/clamav/scanner.go

```diff
@@ -0,0 +1,60 @@
+package clamav
+
+import (
+	"bytes"
+	"fmt"
+
+	clamd "github.com/dutchcoders/go-clamd"
+	"github.com/fsouza/go-dockerclient"
+
+	"github.com/openshift/image-inspector/pkg/api"
+)
+
+type ClamScanner struct {
+	// Socket is the location of the clamav socket.
+	Socket string
+}
+
+var _ api.Scanner = &ClamScanner{}
+
+func NewScanner(socket string) api.Scanner {
+	return &ClamScanner{
+		Socket: socket,
+	}
+}
+
+// Scan will scan the image
+func (s *ClamScanner) Scan(path string, image *docker.Image) error {
+	scanner := clamd.NewClamd(s.Socket)
```

Comment:
I believe this actually creates the socket so I'm not sure.  I noticed that when I was running the daemon with the same socket and started my test scan the daemon actually quit because something else started using the socket.  Definitely an implementation detail that needs worked out.